config/sitemap.rb

```diff
@@ -0,0 +1,36 @@
+SitemapGenerator::Sitemap.default_host = "http://trase.earth"
+SitemapGenerator::Sitemap.create do
+  add('/',
+    :video => {
+      :thumbnail_loc => 'https://i.ytimg.com/vi/wMnAQJBptj8/hqdefault.jpg?sqp=-oaymwEZCNACELwBSFXyq4qpAwsIARUAAIhCGAFwAQ==&rs=AOn4CLDvbL8FkheZ4W98fX-lzgG2z6snlA',
+      :title => 'Introducing Trase',
+      :description => 'Trase is a supply chain transparency initiative that transforms our understanding of globally traded agricultural commodities.',
+      :content_loc => 'https://www.youtube.com/embed/wMnAQJBptj8?autoplay=0&controls=0&disablekb=1&playsinline=1&cc_load_policy=0&cc_lang_pref=auto&widget_referrer=https%3A%2F%2Ftrase.earth%2F%3F&noCookie=false&rel=0&showinfo=0&iv_load_policy=3&modestbranding=1&enablejsapi=1&origin=https%3A%2F%2Ftrase.earth&widgetid=1',
+      :tags => %w[trase supply chain transparency SEI Global Canopy],
+      :category => 'Nonprofits & Activism',
+    },
+    :changefreq => 'monthly'
+  )
+
+  add'/explore', :changefreq => 'monthly'
+  add'/flows', :changefreq => 'monthly'
+  add'/profiles', :changefreq => 'monthly'
+  contexts = Api::V3::Context.all;
+  contexts.each do |context|
```

Comment:
It is slow because it runs too many queries, there are a few things that can be improved:
1. looping over years should be inside looping over node ids, otherwise you're running the exact same queries for each year
2. nodes query needs to be filtered by context id
3. we can halve the number of queries by filtering by `profile IN ('place', 'actor')` and then dynamically inserting the current profile type into the url
4. we can use `pluck` instead of `map` to reduce the number of columns fetched from the database

So in the end we could have something like:
```
  contexts.each do |context|
    Api::V3::Readonly::Node.
      where(context_id: context.id).
      where(profile: ['place', 'actor']).
      pluck(:id, :profile).
      each do |nodeId, profile|
        context.years.each do |year|
          add "/profile-#{profile}?nodeId=#{nodeId}&year=#{year}&contextId=#{context.id}", :changefreq => 'monthly'
        end
      end
  end
```

And it’s quite quick!
Also, perhaps there's room to simplify even more and only include the last year?